### PR TITLE
Move mocha tests - Closes #2847

### LINKED ISSUE #2847: Move Mocha tests to a separate folder
### Expected behavior
all files under framework/test/\*\*/* folder must be moved to /framework/test/mocha/\*\*/* to differentiate jest tests from mocha tests.
### Which version(s) does this affect? (Environment, OS, etc...)
1.6+

### PR DESCRIPTION
**Pull request by yatki**
### What was the problem?
Since we are switching to jest, old mocha tests had to be reorganised.
### How did I fix it?

I moved old mocha tests & mocha.opts file & setup.js file to framework/test/mocha folder. 

This pr contains only file path refactorings.

### Review checklist

* The PR resolves #2847 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

**Comment by nazarhussain** [comment]
@yatki Please list down the issue under this specific category here https://github.com/LiskHQ/lisk/issues/2848 just to keep track of stuff.

File: .gitignore [hunk 1]
```diff
@@ -31,13 +31,12 @@ Session.vim
 
 # Config files
 sftp-config.json
-framework/test/network/pm2.network.json
-framework/test/network/configs/
-framework/test/network/networkTestsLogger.logs
+framework/test/mocha/network/pm2.network.json
+framework/test/mocha/network/configs/
+framework/test/mocha/network/networktest/mocha/sLogger.logs
```
Comment (ManuGowda) [inline .gitignore]: `sLogger.logs`? wired file naming. is it dynamically generated file? proper file name convention will be a good start.

Comment (yatki) [reply .gitignore]: Ah good catch, thanks.

**Comment by yatki** [comment]
> @yatki Please list down the issue under this specific category here #2848 just to keep track of stuff.

@nazarhussain Done.

**Comment by MaciejBaj** [comment]
Please resolve the conflicts.

**Pull request merged**